### ISSUE TITLE
Allow references inside lamda

### ISSUE THREAD

**Issue by pauliusuza** [nick-thompson/blueprint#29]
I'm blocked by `appRoot->engine.registerNativeMethod` not allowing to pass variables inside lambda by reference: `[&](void* stash, const juce::var::NativeFunctionArgs& args)` - note the `[&](...`

I'd like to reference the cast of my processor inside the lambda function. Would it be possible to add support for this use case?

Great Library. Thanks.

**Comment by nick-thompson** [comment]
Hey @pauliusuza, sorry for the delay here, I'm not sure why but I'm not getting notifications for new issues... need to figure that out.

That's right, yea, the native methods need to conform to a c-style function pointer interface because of how duktape works with function calls. Lambdas without a capture group can be cast as such, but as soon as you capture, it breaks the cast. The workaround is to use that stash argument:

```cpp
struct MyData {
    RefToMyThing& thing1;
    RefToMyOtherThing& thing2;
};

MyData myStash { x, y };

appRoot->engine.registerNativeMethod("test", [](void* stash, const juce::var::NativeFunctionArgs& args) {
    auto* myStash_ = reinterpret_cast<MyData*>(stash);
    RefToMyThing& thing1 = myStash->thing1;
    RefToMyOtherThing& thing2 = myStash->thing2;

    ...
}, &myStash);
```

Of course if you follow this example you have to be very careful with your object lifetimes. Let me know if that works for you!

**Comment by nick-thompson** [comment]
Resolved in e3d5057
